element_optogenetics/optogenetics.py

```diff
@@ -1,159 +1,234 @@
+        frequency ( decimal(5, 1) ): Frequency in Hz of the waveform
+        duration ( decimal(5, 1) ): Duration in ms of each optostimulus
+        protocol_description ( varchar(255) ): Protocol description
+    """
+
     definition = """
-    # OptoProtocol defines a single opto stimulus repeat
-    opto_protocol_id     : smallint
+    # Protocol defines a single opto stimulus that repeats
+    protocol_id     : smallint
     ---
     -> OptoWaveform
     -> Device
-    opto_wavelength      : smallint         # (nm) wavelength of the photo stimulation light
-    opto_power           : decimal(6, 2)    # (mW) total power coming out of the light source
-    opto_frequency       : decimal(5, 1)    # (Hz) frequency of waveform
-    opto_duration        : decimal(5, 1)    # (ms) duration of each optostimulus
-    opto_protocol_description='' : varchar(255)  # description of optogenetics protocol
+    wavelength      : smallint              # (nm) wavelength of photo stim. light
+    power           : decimal(6, 2)         # (mW) total power from light source
+    frequency       : decimal(5, 1)         # (Hz) frequency of the waveform
+    duration        : decimal(5, 1)         # (ms) duration of each optostimulus
+    protocol_description='' : varchar(255)  # description of optogenetics protocol
     """
 
 
 @schema
-class OptoSession(dj.Manual):
+class SessionOptoProtocol(dj.Manual):
+    """Session protocol
+
+    Attributes:
+        Session (foreign key): Session primary key
+        OptoProtocol (foreign key): OptoProtocol primary key
+    """
+
     definition = """
     -> Session
+    -> OptoProtocol
     """
 
-    class Protocol(dj.Part):
-        definition = """
-        -> master
-        -> OptoProtocol
-        """
-
-    class BrainRegion(dj.Part):
-        definition = """
-        -> master
-        -> BrainRegion
-        ---
-        light_intensity   : decimal(6, 2)  # (mW/mm2) light intensity at each brain region
-        """
-
-    class BrainLocation(dj.Part):
-        definition = """
-        -> master
-        location_id : int
-        ---
-        ap_location : decimal(6, 2) # (um) anterior-posterior; ref is 0; more anterior is more positive
-        ml_location : decimal(6, 2) # (um) medial axis; ref is 0 ; more right is more positive
-        depth       : decimal(6, 2) # (um) manipulator depth relative to surface of the brain (0); more ventral is more negative
-        theta       : decimal(5, 2) # (deg) - elevation - rotation about the ml-axis [0, 180] - w.r.t the z+ axis
-        phi         : decimal(5, 2) # (deg) - azimuth - rotation about the dv-axis [0, 360] - w.r.t the x+ axis
-        -> BrainRegion
-        light_intensity : decimal(6, 2) # (mW/mm2) light intensity at each brain location
-        """
-
 
 @schema
-class OptoTrial(dj.Imported):
-    definition = """
-    -> SessionTrial
-    -> OptoSession
+class SessionOptoBrainLocation(dj.Manual):
+    """Session brain location
```

Comment:
Is this the location a optic fiber? I'm not sure why this depends on Session since it won't change from session to session?

Reply:
That's a good point. I just met with Thinh to discuss moving a handful of items to Element Animal as a surgery schema following the sabatini lab's [reference schema](https://github.com/bernardosabatinilab/sabatini-datajoint-pipeline/blob/main/workflow/pipeline/reference.py) as a model. I'll keep you posted